node/client/benches/client_bench.rs

```diff
@@ -0,0 +1,363 @@
+use bencher::{benchmark_group, benchmark_main, Bencher};
+
+use client::{BlockProductionResult, ChainConsensusBlockBody, Client};
+use configs::{ChainSpec, ClientConfig};
+use primitives::aggregate_signature::{BlsPublicKey, BlsSecretKey};
+use primitives::block_traits::SignedBlock;
+use primitives::chain::{ChainPayload, ReceiptBlock};
+use primitives::hash::CryptoHash;
+use primitives::signature::{sign, SecretKey as SK, DEFAULT_SIGNATURE};
+use primitives::test_utils::get_key_pair_from_seed;
+use primitives::transaction::{
+    CreateAccountTransaction, DeployContractTransaction, FinalTransactionStatus,
+    FunctionCallTransaction, SendMoneyTransaction, SignedTransaction, TransactionBody,
+};
+use primitives::types::{MessageDataBody, SignedMessageData};
+use serde_json::Value;
+use std::collections::{HashMap, HashSet};
+use std::path::Path;
+
+const TMP_DIR: &str = "./tmp_bench/";
+const ALICE_ACC_ID: &str = "alice.near";
+const BOB_ACC_ID: &str = "bob.near";
+const CONTRACT_ID: &str = "contract.near";
+const DEFAULT_BALANCE: u64 = 10_000_000;
+const DEFAULT_STAKE: u64 = 1_000_000;
+
+fn get_bls_keys(seed: [u32; 4]) -> (BlsPublicKey, BlsSecretKey) {
+    use rand::{SeedableRng, XorShiftRng};
+    let mut rng = XorShiftRng::from_seed(seed);
+    let secret_key = BlsSecretKey::generate_from_rng(&mut rng);
+    let public_key = secret_key.get_public_key();
+    (public_key, secret_key)
+}
+
+fn get_chain_spec() -> (ChainSpec, SK, SK) {
```

Comment:
I feel we already have get_chain_spec in some other test

Reply:
`ChainSpec` is a config object. I think it is totally normal when different tests construct different configs that suit their purpose. It is also better than adding dependencies between the crates.